File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/events/OutboxJobSummaryResponse.java
package com.mieai.qqbot.admin.events;

import java.time.Instant;
import java.util.UUID;

/** Public, non-payload representation used by Outbox and DLQ lists. */
public record OutboxJobSummaryResponse(
        UUID id,
        String environment,
        UUID botId,
        String botDisplayName,
        String appId,
        UUID sourceEventId,
        String jobType,
        String status,
        long attempt,
        Instant availableAt,
        Instant createdAt,
        Instant updatedAt,
        Instant completedAt,
        String lastError) {
}

File: qqbot-module-operations/src/main/kotlin/com/mieai/qqbot/admin/events/OutboxJobSummaryResponse.kt
package com.mieai.qqbot.admin.events

import java.time.Instant
import java.util.UUID

/** Public, non-payload representation used by Outbox and DLQ lists. */
data class OutboxJobSummaryResponse(
    val id: UUID,
    val environment: String,
    val botId: UUID,
    val botDisplayName: String?,
    val appId: String?,
    val sourceEventId: UUID?,
    val jobType: String,
    val status: String,
    val attempt: Long,
    val availableAt: Instant,
    val createdAt: Instant,
    val updatedAt: Instant,
    val completedAt: Instant?,
    val lastError: String?,
)
